GherkinSyncTool.Synchronizers.AllureTestOps/AllureTestOpsSynchronizer.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            var stopwatch = Stopwatch.StartNew();
+            Log.Info("# Start synchronization with Allure TestOps");
+
+            var allureTestCases = _allureClient.GetAllTestCases().ToList();
+            
+            foreach (var featureFile in featureFiles)
+            {
+                var insertedTagIds = 0;
+                
+                foreach (var scenario in featureFile.Document.Feature.Children.OfType< Scenario>())
+                {
+                    var tagId = scenario.Tags.FirstOrDefault(tag => tag.Name.Contains(_gherkinSyncToolConfig.TagIdPrefix));
+
+                    var caseRequest = _caseContentBuilder.BuildCaseRequest(scenario, featureFile);
+
+                    // Create test case for feature file which is getting synced for the first time, so no tag id present.  
+                    if (tagId is null)
+                    {
+                        TestCase addCase;
+                        try
+                        {
+                            addCase = _allureClient.AddTestCase(caseRequest);
+                        }
+                        catch (AllureException e)
+                        {
+                            Log.Error(e, $"The test case has not been created: {scenario.Name}");
+                            _context.IsRunSuccessful = false;
+                            continue;
+                        } 
+                        
+                        var lineNumberToInsert = scenario.Location.Line - 1 + insertedTagIds;
+                        var formattedTagId = GherkinHelper.FormatTagId(addCase.Id.ToString());
+                        TextFilesEditMethods.InsertLineToTheFile(featureFile.AbsolutePath, lineNumberToInsert,
+                            formattedTagId);
+                        insertedTagIds++;
+                    }
+
+                    // Update scenarios that have tag id
+                    if (tagId is not null)
+                    {
+                        var caseIdFromFile = GherkinHelper.GetTagId(tagId);
+                        //featureFilesTagIds.Add(caseId);
+                        var allureTestCase = allureTestCases.FirstOrDefault(c => c.Id == caseIdFromFile);
+                        if (allureTestCase is null)
+                        {
+                            Log.Warn($"Case with id {caseIdFromFile} not found. Recreating missing case");
+                            TestCase addCase;
+                            try
+                            {
+                                addCase = _allureClient.AddTestCase(caseRequest);
+                            }
+                            catch (AllureException e)
+                            {
+                                Log.Error(e, $"The case has not been created: {scenario.Name}");
+                                _context.IsRunSuccessful = false;
+                                continue;
+                            }
```

Comment:
Looks like lines 51 - 62 and 78 - 90 are the same. It calls for method extraction

Reply:
Done